google/auth/credentials.py

```diff
@@ -154,6 +186,20 @@ def apply(self, headers, token=None):
         if self.quota_project_id:
             headers["x-goog-user-project"] = self.quota_project_id
 
+    def _blocking_refresh(self, request):
+        if not self.valid:
+            self.refresh(request)
+
+    def _non_blocking_refresh(self, request):
+        if (
```

Comment:
following the comment above, this would become:

```
if self.token_state == TokenState.FRESH:
  return

need_refresh = True
if self.token_state == TokenState.STALE:
  need_refresh = not self.refresh_worker.start_refresh(self, request)

if need_refresh:
  self.refresh(request)
  self._refresh_worker.clear_error()

Reply:
Done! I kept the `INVALID` case, should a refresh was never called in the STALE state.